File: hyesu/그래프탐색/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    queue = deque()
    visited = [False] * n
    
    for i in range(n):
        print(visited)
        if visited[i] == False:
            answer += 1
            queue.append(i)
            visited[i] = True
            
            while queue:
                v = queue.popleft()
                for j in range(n):
                    if computers[v][j] == 1 and visited[j] == False:
                        queue.append(j)
                        visited[j] = True

    return answer

File: hyesu/SwiftAlgorithm/SwiftAlgorithm/네트워크.swift
//
//  네트워크.swift
//  SwiftAlgorithm
//
//  Created by 김혜수 on 2023/02/15.
//

import Foundation

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    var visited = [Bool](repeating: false, count: n)
    var queue: [Int] = []
    var v = 0
    var answer = 0
    
    for idx in visited.indices {
        if visited[idx] == false {
            answer += 1
            queue.append(idx)
            visited[idx] = true
        }

        while queue.count > 0 {
            v = queue.removeLast()
            for i in 0..<n {
                if computers[v][i] == 1 && visited[i] == false {
                    queue.append(i)
                    visited[i] = true
                }
            }
        }
    }
    
    return answer
}
